src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -225,14 +230,28 @@ public Response addGzip(InputStream gzip)
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
 			return builder.addError(e.getMessage()).build();
 		}
-		return (add(gzipInputStream));
+		return (add(httpHeaders, gzipInputStream));
 	}
 
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path("/datapoints")
-	public Response add(InputStream json)
+	public Response add(@Context HttpHeaders httpHeaders, InputStream json)
 	{
+//		ScopeManager scopeManager = tracer.scopeManager();
+//		SpanContext spanContext = tracer.extract(Format.Builtin.HTTP_HEADERS, new HttpHeadersCarrier(httpHeaders.getRequestHeaders()));
+//		Tracer.SpanBuilder spanBuild = tracer.buildSpan("/datapoints").withTag("Name","datapoints").withTag(Tags.SPAN_KIND.getKey(),Tags.SPAN_KIND_SERVER);
+//		if (spanContext != null)
+//			spanBuild.asChildOf(spanContext);
+//
+//		Span span = spanBuild.start();
+//		Scope scope = scopeManager.activate(span, true);
+//		span.log(json.toString());
+
+		Scope scope = activateScope("datapoints", httpHeaders);
+		Span span = scope.span();
+		span.log(json.toString());
```

Comment:
That will be very noisy and will explode in volume. I suggest removing this log.